### PR TITLE
Add new doc for deploying third part app

### PR DESCRIPTION
**Pull request by Aisuko**
This PR is related to #3973 
And the PR has linked to the `Kustomize docs migration coordination` Sheet No.12. @KnVerey Feel free to give feedback. Thank you.

### CODE REVIEW

File: examples/thirdPartApp/base/service.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+kind: Service
+metadata:
+  name: example-app
+spec:
+  selector:
+    app: example-app
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 80
```
Comment (koba1t) [inline examples/thirdPartApp/base/service.yaml:11]: Could you add a newline?

File: examples/thirdPartApp/overlays/production/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#               - name: config-volume
+#                 configMap:
+#                   name: example-app-configmap
+#               - name: secrets-volume
+#                 secret:
+#                   secretName: example-app-secrets
+#     target:
+#       kind: Deployment
+#       name: example-app
+#       version: apps/v1
```
Comment (koba1t) [inline examples/thirdPartApp/overlays/production/kustomization.yaml:49]: Looks like `version` must be set to `v1`. I think you have to write  `apps` at `group` literally if you need.
Please watch here: https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/#patch-using-path-json6902

File: examples/thirdPartApp/overlays/tests/configmap.yaml
Comment (koba1t) [inline examples/thirdPartApp/overlays/tests/configmap.yaml:1]: Are you need these files?
It looks like every configMap and secret will be generated by kustomize.

**Review by koba1t — CHANGES_REQUESTED** [review]

**Comment by koba1t** [comment]
Hi @Aisuko 

Thanks for your contribution!
I apologize for the super delay to review.

@KnVerey is still very busy, check this PR instead.
I already put in some comments. Could you check that?

File: examples/thirdPartApp/overlays/production/kustomization.yaml (hunk 2 above)
Comment (Aisuko) [reply examples/thirdPartApp/overlays/production/kustomization.yaml:49]: Nice catch

File: examples/thirdPartApp/overlays/tests/configmap.yaml
Comment (Aisuko) [reply examples/thirdPartApp/overlays/tests/configmap.yaml:1]: These are demo files still useful.

File: examples/thirdPartApp/base/service.yaml (hunk 1 above)
Comment (Aisuko) [reply examples/thirdPartApp/base/service.yaml:11]: Sure.

**Comment by Aisuko** [comment]
> Hi @Aisuko
> 
> Thanks for your contribution! I apologize for the super delay to review.
> 
> @KnVerey is still very busy, check this PR instead. I already put in some comments. Could you check that?

Please do not say sorry for that, and thanks for your review. By the way, I saw there many of the sig-group members are so busy, I saw many of the PRs cannot get a review and merge. So, may I join the `kubernetes-sig` community as a member and contribute to and maintain these projects which need more maintainers?

**Comment by koba1t** [comment]
> Please do not say sorry for that, and thanks for your review. By the way, I saw there many of the sig-group members are so busy, I saw many of the PRs cannot get a review and merge. So, may I join the kubernetes-sig community as a member and contribute to and maintain these projects which need more maintainers? And it is my pleasure if I can get help from you on joining the community.

We are very happy if you will continue contributing and joining the community!
You can check any PRs and when the PR looks good to you, you can leave a `/lgtm` comment now. Your review is helpful to add `approve` from the project maintainer.
I told to the maintainer of kustomize and with pleasure sponsorshp with you after you review more a few PRs.
If you are interesting other projects, Please contact the relevant sigs maintainers after you review a few PRs due to suponsor are limited your active sig members!

File: examples/thirdPartApp/overlays/production/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    # behavior: merge
+    literals:
+      - DATABASE_URL=production_database_url
+      - API_KEY=production_api_key
+secretGenerator:
+  - name: example-app-secrets
+    # behavior: merge
+    literals:
+      - DB_PASSWORD=production_data_password
+# patches does not working right here, maybe can open an issue.
```
Comment (koba1t) [inline examples/thirdPartApp/overlays/production/kustomization.yaml:16]: Could you remove this comment?

File: examples/thirdPartApp/overlays/production/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                  name: example-app-configmap
+              - name: secrets-volume
+                secret:
+                  secretName: example-app-secrets
+    target:
+      group: apps
+      kind: Deployment
+      name: example-app
+      version: v1
+patchesStrategicMerge:
```
Comment (koba1t) [inline examples/thirdPartApp/overlays/production/kustomization.yaml:51]: Are you still need these lines?

File: examples/thirdPartApp/overlays/tests/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  - ../../base
+configMapGenerator:
+  - name: example-app-config
+    literals:
+      - APP_ENV=testing
+secretGenerator:
+  - name: example-app-secrets
+    literals:
+      - DB_PASSWORD=password
+patchesStrategicMerge:
```
Comment (koba1t) [inline examples/thirdPartApp/overlays/tests/kustomization.yaml:13]: I think `patchesStrategicMerge` is now deprecated. 
Could you use [patches](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/) syntax?

**Review by koba1t — CHANGES_REQUESTED** [review]
Thanks for your continued contribution, @Aisuko 
I have added my review for this PR.
Could you check my comments?

**Comment by Aisuko** [comment]
@koba1t Thanks for your suggestions. I will help review more PRs.

And I have tested the latest version of code of this PR, it should be ok.
```
@Aisuko ➜ /workspaces/kustomize (feat/doc) $ kubectl kustomize examples/thirdPartApp/overlays/production/
apiVersion: v1
data:
  API_KEY: production_api_key
  DATABASE_URL: production_database_url
kind: ConfigMap
metadata:
  name: example-app-configmap-5g8mm776bk
---
apiVersion: v1
data:
  DB_PASSWORD: cHJvZHVjdGlvbl9kYXRhX3Bhc3N3b3Jk
kind: Secret
metadata:
  name: example-app-secrets-97tk7dc6f9
type: Opaque
---
apiVersion: v1
kind: Service
metadata:
  name: example-app
spec:
  ports:
  - port: 80
    protocol: TCP
    targetPort: 80
  selector:
    app: example-app
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: example-app
spec:
  replicas: 5
  selector:
    matchLabels:
      app: example-app
  template:
    metadata:
      labels:
        app: example-app
    spec:
      containers:
      - env:
        - name: ENVIRONMENT
          value: production
        image: example-app:latest
        name: example-app
        ports:
        - containerPort: 80
        volumeMounts:
        - mountPath: /etc/config
          name: config-volume
          readOnly: true
        - mountPath: /etc/secrets
          name: secrets-volume
          readOnly: true
      volumes:
      - configMap:
          name: example-app-configmap-5g8mm776bk
        name: config-volume
      - name: secrets-volume
        secret:
          secretName: example-app-secrets-97tk7dc6f9
```

File: examples/thirdPartApp/overlays/production/kustomization.yaml (hunk 3 above)
Comment (Aisuko) [reply examples/thirdPartApp/overlays/production/kustomization.yaml]: Done

File: examples/thirdPartApp/overlays/production/kustomization.yaml (hunk 4 above)
Comment (Aisuko) [reply examples/thirdPartApp/overlays/production/kustomization.yaml]: Already removed it.

File: examples/thirdPartApp/overlays/tests/kustomization.yaml (hunk 5 above)
Comment (Aisuko) [reply examples/thirdPartApp/overlays/tests/kustomization.yaml:13]: Done

**Comment by koba1t** [comment]
@Aisuko 
Thanks! Looks like to me!
/lgtm

/cc @natasha41575 @annasong20

**Review by kundan2707 — COMMENTED** [review]
/lgtm

**Comment by Aisuko** [comment]
Hi @annasong20. May you please help to merge this one?

**Comment by Aisuko** [comment]
Hi @natasha41575 May you please to help to merge this one?

**Comment by natasha41575** [comment]
Do you mind elaborating more here on what this example does, and what sort of workflow it walks the user through? This should also be explained in the example's README.

At first glance, this appears to be a basic kustomization directory that uses the `resources, `secretGenerator`, `configMapGenerator` and `patches` fields. A lot of these are already covered in [some of the other examples](https://github.com/kubernetes-sigs/kustomize/tree/master/examples) that we have, so it would be good to have context on how this is different from the other examples that we have. It seems pretty similar to the [hello-world example](https://github.com/kubernetes-sigs/kustomize/tree/master/examples/helloWorld).

Another thing is that this doesn't quite follow the same format as our other examples - those have a much more in-depth readme that walk you through how to create your own kustomization directory and add files to it. For example, you can look at the [hello-world example](https://github.com/kubernetes-sigs/kustomize/tree/master/examples/helloWorld) and see that it explains every step of the process and what each file does. We would like to keep all our examples up to that same level of thoroughness and explanation.

**Pull request closed without merging**